src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -159,9 +159,7 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
             // this is a no-op.  Emitting an error here, or when the original parameter was bound, would
             // adversely effect the compilation or potentially change overload resolution.  
             var compilation = this.DeclaringCompilation;
-            if (Type.ContainsDynamic() && 
-                compilation.HasDynamicEmitAttributes() &&
-                compilation.GetSpecialType(SpecialType.System_Boolean).SpecialType == SpecialType.System_Boolean)
```

Comment:
If I remember correctly, Jared's intent was to ensure boolean type is available and is good (similar to how it is done in Binder.ReportUseSiteDiagnosticForDynamic). Simply removing this condition wouldn't be the right thing to do, instead we should change it to check that symbol for the boolean type doesn't have use-site error (use-site warning is Ok).